packages/react/src/components/DateTimePicker/DateTimePicker.test.jsx

```diff
@@ -685,6 +680,40 @@ describe('DateTimePicker', () => {
     expect(applyBytton).toBeDisabled();
   });
 
+  it('should disable apply button when absolute TimePickerSpinner input is invalid', () => {
+    const { i18n } = DateTimePicker.defaultProps;
+    render(<DateTimePicker {...dateTimePickerProps} id="picker-test" />);
+    jest.runAllTimers();
+
+    userEvent.click(screen.getByTestId('date-time-picker__field'));
+    userEvent.click(screen.queryByText(DateTimePicker.defaultProps.i18n.customRangeLinkLabel));
+    userEvent.click(screen.queryByText(DateTimePicker.defaultProps.i18n.absoluteLabel));
+    const applyBytton = screen.getByRole('button', { name: i18n.applyBtnLabel });
+
+    // Get start and end time inputs
+    const startTime = screen.getByLabelText('Start time');
+    const endTime = screen.getByLabelText('End time');
+
+    // confirm that start and end time is: 00:00
+    expect(startTime).toHaveValue('00:00');
+    expect(endTime).toHaveValue('00:00');
+
+    // start time and end times are equally initally - hence start time is invalid and apply button is disabled
+    expect(applyBytton).toBeDisabled();
+
+    // Change end time to 01:00 - start time is before end time
+    fireEvent.change(endTime, { target: { value: '01:00' } });
+    expect(applyBytton).toBeEnabled();
+
+    // Change start time to 02:00 - start time is after end time
+    fireEvent.change(startTime, { target: { value: '02:00' } });
+    expect(applyBytton).toBeDisabled();
+
+    // Change end time to 02:00 - start time is after end time
+    fireEvent.change(endTime, { target: { value: '00:00' } });
+    expect(applyBytton).toBeDisabled();
+  });
+
```

Comment:
Can you add a little more testing that changes the date and confirms situations where the start time before an end time is valid? ie. start: `02:00` and end: `01:00` is valid if startDate is `2021/10/01` and endDate is `2021/11/01`

Reply:
The date wouldn't matter because the date picker will always set the start date as the first date selected and set the last date as the second selected date. So the test mentioned is already covered in the current test cases.

Reply:
Thanks, @eosineye1, but unfortunately, this isn't correct. The DateTimePicker can be passed a selected date range, so these tests aren't covering the validation correctly. See this [existing story in your preview](https://deploy-preview-3112--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-datetime-picker--selected-absolute). It marks the startTime as invalid, but it _is_ a valid time.